projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -32,21 +32,116 @@ import { VirtualHelperComponent } from './virtual.helper.component';
 
 @Directive({ selector: '[igxFor][igxForOf]' })
 export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestroy {
-    @Input() public igxForOf: any[];
-    @Input() public igxForScrollOrientation: string;
-    @Input() public igxForScrollContainer: any;
-    @Input() public igxForContainerSize: any;
-    @Input() public igxForItemSize: any;
+
+    /**
+     * An @Input property that sets the data to be rendered.
+     * ```html
+     * <ng-template igxFor let-item [igxForOf]="data"></ng-template>
+     * ```
+     */
+    @Input()
+    public igxForOf: any[];
+
+    /**
+     * An @Input property that specifies the scroll orientation.
+     * Only `vertical` and `horizontal` are valid.
+     * ```html
+     * <ng-template igxFor let-item [igxForOf]="data" [igxForScrollOrientation]="'horizontal'"></ng-template>
+     * ```
+     */
+    @Input()
+    public igxForScrollOrientation: string;
+
+    /**
+     * Optionally pass the parent `igxFor` instance to create a virtual template scrolling both horizontally and vertically.
+     * ```html
+     * <ng-template #scrollContainer igxFor let-rowData [igxForOf]="data"
+     *       [igxForScrollOrientation]="'vertical'"
+     *       [igxForContainerSize]="'500px'"
+     *       [igxForItemSize]="'50px'"
+     *       let-rowIndex="index">
+     *       <div [style.display]="'flex'" [style.height]="'50px'">
+     *           <ng-template #childContainer igxFor let-item [igxForOf]="data"
+     *               [igxForScrollOrientation]="'horizontal'"
+     *               [igxForScrollContainer]="parentVirtDir"
+     *               [igxForContainerSize]="'500px'">
+     *                   <div [style.min-width]="'50px'">{{rowIndex}} : {{item.text}}</div>
+     *           </ng-template>
+     *       </div>
+     * </ng-template>
+     * ```
+     */
+    @Input()
+    public igxForScrollContainer: any;
+
+    /**
+     * An @Input property that sets the px-affixed size of the container along the axis of scrolling.
+     * ```html
+     * <ng-template igxFor let-item [igxForOf]="data" [igxForContainerSize]="'500px'"></ng-template>
```

Comment:
@alexalex972 if `igxForScrollOrientation` is mandatory for igxFor to work then you should use it in every code snippet.